src/frequenz/client/base/streaming.py

```diff
@@ -130,27 +181,77 @@ def __init__(  # pylint: disable=too-many-arguments,too-many-positional-argument
         )
         self._retry_on_exhausted_stream = retry_on_exhausted_stream
 
-        self._channel: channels.Broadcast[StreamEvent | OutputT] = channels.Broadcast(
-            name=f"GrpcStreamBroadcaster-{stream_name}"
+        # Channel for transformed data messages (OutputT)
+        self._data_channel: channels.Broadcast[OutputT] = channels.Broadcast(
+            name=f"GrpcStreamBroadcaster-{stream_name}-Data"
         )
+
+        # Channel for stream events (StreamEvent), created on demand
+        self._event_channel: channels.Broadcast[StreamEvent] | None = None
+        self._event_sender: channels.Sender[StreamEvent] | None = None
         self._task = asyncio.create_task(self._run())
 
+    @overload
+    def new_receiver(
+        self,
+        maxsize: int = 50,
+        warn_on_overflow: bool = True,
+        *,
+        include_events: Literal[False] = False,
+    ) -> channels.Receiver[OutputT]: ...
+
+    @overload
+    def new_receiver(
+        self,
+        maxsize: int = 50,
+        warn_on_overflow: bool = True,
+        *,
+        include_events: Literal[True],
+    ) -> channels.Receiver[StreamEvent | OutputT]: ...
+
     def new_receiver(
-        self, maxsize: int = 50, warn_on_overflow: bool = True
-    ) -> channels.Receiver[StreamEvent | OutputT]:
+        self,
+        maxsize: int = 50,
+        warn_on_overflow: bool = True,
+        *,
```

Comment:
Not really part of this PR, but since you are adding the `*` here it makes me think that we should probably put all other arguments as keyword-only, as it can't be really inferred what they are by the type.